### PR TITLE
SA deprecation notice for April 15 GA

### PR DESCRIPTION
**Pull request by pohannigansfx**
Deprecation notice in documentation for Smart Agent -- **please do not merge until April 15.**  PR includes files with a note about deprecation that link back to new Deprecation Notice page. This is only for files in the signalfx-agent repo.

### CODE REVIEW

**Comment by pohannigansfx** [comment]
PR now includes all files in this repo that should have Smart Agent deprecation notice for Olly GA

File: README.md [hunk 1]
```diff
@@ -3,6 +3,8 @@
 [![GoDoc](https://godoc.org/github.com/signalfx/signalfx-agent?status.svg)](https://godoc.org/github.com/signalfx/signalfx-agent)
 [![CircleCI](https://circleci.com/gh/signalfx/signalfx-agent.svg?style=shield)](https://circleci.com/gh/signalfx/signalfx-agent)
 
+SignalFx Smart Agent is deprecated. For details, see the [Deprecation Notice](/docs/smartagent-deprecation-notice).
```
Comment (keitwb) [inline README.md:6]: Don't you think this should have some bolding to make it stand out a bit more?

Same in all the other places.

Comment (pohannigansfx) [reply README.md:6]: It's first line of the topic, right under the title. Not vivid enough?

Comment (flands) [reply README.md:6]: More vivid the better see: https://github.com/signalfx/gateway#deprecation-notice. In fact, would be best to have everything in doc in main readme and all links point to main readme.

**Review by gschatz-splunk — APPROVED** [review]

File: docs/smartagent-deprecation-notice.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+# Deprecation Notice for SignalFx Smart Agent
+
+Smart Agent is deprecated in conjunction with the release of Splunk Observability Cloud:
+
+* Smart Agent continues to function and is still supported by Splunk.
+* Apart from the [documented exceptions](https://docs.splunk.com/Observability/get-started/migrate/migrate-to-otel.html) where integration with Splunk Observability Cloud still requires deploying Smart Agent, Smart Agent functionality is replaced by the [Splunk Distribution of OpenTelemetry Connector](https://docs.splunk.com/Observability/get-started/get-data-in/get-data-in.html).
+* Splunk is committed to further development of OpenTelemetry, and might remove Smart Agent in a future version.
```
Comment (jadbSplunk) [inline docs/smartagent-deprecation-notice.md:7]: I'm not sure what "might remove Smart Agent in a future version" means. Suggest "may discontinue support for Smart Agent in the future and will provide customers advance notice of such discontinuation."

Comment (pohannigansfx) [reply docs/smartagent-deprecation-notice.md]: Third bullet now says this:
"Splunk is committed to further development of OpenTelemetry, and may discontinue support for Smart Agent in the future after providing customers advance notice of such discontinuation."

**Comment by pohannigansfx** [comment]
Approved by Principal PM -- please merge this into the repo.

**Comment by asuresh4** [comment]
@pohannigansfx - I pushed some updates to fix tests. I took a look and I believe your original changes are intact. PTAL

**Comment by pohannigansfx** [comment]
@asuresh4 you are correct. If you don't mind a final force push, I revised one line just a bit -- but otherwise LGTM.

**Comment by pohannigansfx** [comment]
Thanks, @asuresh4. We good to go and merge this thing? (I don't have merge permissions in this repo)

**Review by asuresh4 — APPROVED** [review]
The latest failure in `check_links` job seems unrelated to this PR. And the URL actually is accessible.

```
FILE: /usr/src/signalfx-agent/docs/monitors/collectd-php-fpm.md
[✖] https://www.php.net/manual/en/install.fpm.configuration.php#pm.status-path → Status: 503

5 links checked.

ERROR: 1 dead links found!
[✖] https://www.php.net/manual/en/install.fpm.configuration.php#pm.status-path → Status: 503
```

**Pull request merged**